### PR TITLE
Add instructor in training setting for script level on lesson edit page

### PR DESCRIPTION
**Pull request by dmcavoy**
Beginning of the work to create a way for participants in a PL course to view parts of a level that are typical reserved for instructors only. This PR adds the new property `instructor_in_training` to ScriptLevels and then creates an editing interface in the LevelDetailsToken to set this field. This field will only be able to be set on professional learning courses (any course where participant_audience is not students). However for the time being while we have not finished setting `participant_audiences`, `professionalLearningCourse` has been set to always be `false` which means this field can't be used yet so should be safe to deploy.

Screen Shot 2021-10-27 at 3 42 21 PM
Screen Shot 2021-10-27 at 3 42 27 PM

I just messed with the props locally so I could show what the purple indicator will look like on the LevelToken:

Screen Shot 2021-10-27 at 3 49 17 PM

## Links

- [Part 6 of Eng Plan](https://docs.google.com/document/d/1V61xONU0-mleMEEdHWNxhvZtYIPRCB31mT_hbUXusCQ/edit#heading=h.388ez5k6ldev)
- [Req 3 of Product Spec](https://docs.google.com/document/d/1qZpc90daxQiiqpcQUiUiGV2tQoaYEQUzOA72UZzBlVE/edit#heading=h.k2rev0wnnwd2)
- [Jira](https://codedotorg.atlassian.net/browse/PLAT-1344)

## Testing story
- Added Unit Tests

### CODE REVIEW

**Comment by davidsbailey** [comment]
Top level feedback -- if there's no exact spec for it already, i think it would be better to hide the "instructor in training" checkbox entirely if we are not in a PL course. the way it is right now seems to be steering those who discover it toward changing the instruction type, which doesn't seem like the right direction for most people working on courses where the participants are students.

**Comment by dmcavoy** [comment]
> Top level feedback -- if there's no exact spec for it already, i think it would be better to hide the "instructor in training" checkbox entirely if we are not in a PL course. the way it is right now seems to be steering those who discover it toward changing the instruction type, which doesn't seem like the right direction for most people working on courses where the participants are students.

Thanks for the feedback Dave. The spec does not deal with this scenario so I just designed it similar to how bonus worked. @tess323 what do you think?

**Comment by tess323** [comment]
Thats a good call @davidsbailey.  My ideal UI would be this feature is available if a script's participant was someone other than a student.  Is that doable while staying in the same location? Sorry for missing this sooner!

**Comment by dmcavoy** [comment]
Professional Learning Course:
Screen Shot 2021-11-01 at 9 55 00 PM

Not Professional Learning Course:
Screen Shot 2021-11-01 at 9 55 58 PM

File: apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx [hunk 1]
```diff
@@ -41,7 +41,13 @@ const createStoreWithoutLessonPlan = () => {
   });
   const store = createStoreWithReducers();
   store.dispatch(
-    init([sampleActivityForLessonWithoutLessonPlan], searchOptions, [], false)
+    init(
+      [sampleActivityForLessonWithoutLessonPlan],
+      searchOptions,
+      [],
+      false,
+      false
```
Comment (davidsbailey) [inline apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx]: this seems like one too many unnamed boolean arguments. can you try turning these into named parameters?
```suggestion
      {
        lessonExtrasAvailableForUnit: false,
        isProfessionalLearningCourse: false
      }
```

File: apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js [hunk 2]
```diff
@@ -37,13 +37,15 @@ export const init = (
   activities,
   searchOptions,
   programmingEnvironments,
-  lessonExtrasAvailableForUnit
+  lessonExtrasAvailableForUnit,
+  isProfessionalLearningCourse
```
Comment (davidsbailey) [inline apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js]: then you can use destructuring syntax here:
```suggestion
  { 
    lessonExtrasAvailableForUnit,
    isProfessionalLearningCourse
  }
```
disclaimer: i haven't tried this out, and all the examples i looked at show ALL parameters being passed inside a single object. so, it's possible you'll have to move all the params inside a single object to get this to work.

File: apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js [hunk 3]
```diff
@@ -624,7 +634,8 @@ export default {
...
```
Comment (davidsbailey) [inline apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js]: this pattern seems odd -- if you are ever writing a new redux reducer, I would suggest exporting a single reducer function instead.

File: apps/test/unit/lib/levelbuilder/lesson-editor/LevelTokenDetailsTest.js [hunk 4]
```diff
@@ -51,20 +51,37 @@ describe('LevelTokenDetails', () => {
...
     };
   });
 
   it('renders with default props', () => {
     const wrapper = shallow(<LevelTokenDetails {...defaultProps} />);
 
-    assertCheckboxVisible(wrapper, 'bonus', true);
-    assertCheckboxVisible(wrapper, 'assessment', true);
-    assertCheckboxVisible(wrapper, 'challenge', true);
+    assertCheckboxVisible(wrapper, 'Bonus', true);
```
Comment (davidsbailey) [inline apps/test/unit/lib/levelbuilder/lesson-editor/LevelTokenDetailsTest.js:62]: I know that "renders with default props" was encouraged at one point, but can you please change the name of this test case to describe what is being tested? e.g. it "renders unchecked checkboxes"

File: apps/src/lib/levelbuilder/lesson-editor/LevelTokenDetails.jsx [hunk 5]
```diff
@@ -129,7 +149,7 @@ export const UnconnectedLevelTokenDetails = LevelTokenDetails;
 
 export default connect(
   state => ({
-    lessonExtrasAvailableForUnit: state.lessonExtrasAvailableForUnit
+    lessonExtrasAvailableForUnit: state.unitInfo.lessonExtrasAvailableForUnit
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/lesson-editor/LevelTokenDetails.jsx]: add isprofessionallearningCourse back here

**Review by davidsbailey — APPROVED** [review]
looks great Dani, thanks for updating!

**Pull request merged**